SQL/sqlRules.json

```diff
@@ -0,0 +1,42 @@
+{
+  "SQL_Server_Auth_EntraIDIntegrated": {
+    "expected": true,
+    "explanation": "The server should be integrated with Azure Active Directory for authentication and authorization. For more information: https://learn.microsoft.com/en-us/sql/relational-databases/security/authentication-access/azure-ad-authentication-sql-server-overview?view=sql-server-ver16",
```

Comment:
This link does not seem to be the right one as it is related to sql server (on-prem) and not Azure SQL. This one seems better: https://learn.microsoft.com/en-us/azure/azure-sql/database/authentication-aad-configure?view=azuresql&tabs=azure-powershell

Also, please change the reference to Azure Active Directory to Microsoft Entra ID instead as it is its new name